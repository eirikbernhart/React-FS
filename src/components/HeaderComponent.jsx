import React, { Component } from 'react';
import catPic from '.././resources/img/404NotFound.jpeg';
import { Link, NavLink } from 'react-router-dom';
import '.././styles/styles.scss';

  



class HeaderComponent extends Component {

    
     render() {
         return (
            <div>
                <header style={{backgroundColor: "white"}}>
                    <p>I'm da HEADER!</p>
                    <NavLink to="/" activeClassName="is-active" exact={true}>Login-page</NavLink>
                    <NavLink to="/register" activeClassName="is-active" exact={true}>Register-page</NavLink>
                    <NavLink to="/main" activeClassName="is-active" exact={true}>Main-page</NavLink>
                </header>              
            </div>
         )
    }


}


export default HeaderComponent;