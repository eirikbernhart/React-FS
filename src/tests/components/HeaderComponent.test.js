import React from 'react';
import {shallow} from 'enzyme';
import HeaderComponent from '../../components/HeaderComponent';
import toJSON from 'enzyme-to-json';


test('should render header correctly', () => {
    const wrapper = shallow(<HeaderComponent />);
    expect(toJSON(wrapper)).toMatchSnapshot();

});