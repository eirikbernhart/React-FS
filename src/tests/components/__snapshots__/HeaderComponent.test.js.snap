// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render header correctly 1`] = `
<div>
  <header
    style={
      Object {
        "backgroundColor": "#eeeedf",
      }
    }
  >
    <NavLink
      activeClassName="is-active"
      ariaCurrent="true"
      className="linkStyle"
      exact={true}
      to="/"
    >
      Login
    </NavLink>
    <NavLink
      activeClassName="is-active"
      ariaCurrent="true"
      className="linkStyle"
      exact={true}
      to="/register"
    >
      Sign Up!
    </NavLink>
    <NavLink
      activeClassName="is-active"
      ariaCurrent="true"
      className="linkStyle"
      exact={true}
      to="/main"
    >
      Main page
    </NavLink>
    <NavLink
      activeClassName="is-active"
      ariaCurrent="true"
      className="linkStyle"
      exact={true}
      to="/inspiration"
    >
      Inspiration page
    </NavLink>
  </header>
</div>
`;
